source/MaterialXGenOgsXml/GlslFragmentGenerator.cpp

```diff
@@ -301,9 +301,19 @@ ShaderPtr GlslFragmentGenerator::generate(const string& fragmentName, ElementPtr
             }
             else
             {
-                if (!outputSocket->getType()->isFloat3())
+                if (context.getOptions().hwTransparency)
+                {
```

Comment:
To match declared return type from line 182.

Reply:
Right on, this came up in my HLSL tests as well, and your fix is better: https://github.com/autodesk-forks/MaterialX/pull/694/files/539c06b77a128d04e43ae91312823f55efd7ca9a